src/main/java/com/exasol/spark/common/FilterConverter.java

```diff
@@ -0,0 +1,218 @@
+
+        case IS_NULL:
+            return BooleanTerm.isNull(column(((IsNull) filter).attribute()));
+
+        case IS_NOT_NULL:
+            return BooleanTerm.isNotNull(column(((IsNotNull) filter).attribute()));
+
+        case IN:
+            final In in = (In) filter;
+            return BooleanTerm.in(column(in.attribute()), getMappedLiteralValues(in.values()));
+
+        case NOT:
+            final Not not = (Not) filter;
+            if (not.child() instanceof In) {
+                final In notIn = (In) not.child();
+                return BooleanTerm.notIn(column(notIn.attribute()), getMappedLiteralValues(notIn.values()));
+            } else if (not.child() instanceof EqualTo) {
+                final EqualTo notEqualTo = (EqualTo) not.child();
+                return BooleanTerm.compare(column(notEqualTo.attribute()), "<>", getLiteralValue(notEqualTo.value()));
+            }
+            final BooleanExpression notResult = convertFilter(not.child());
+            return notResult == null ? null : BooleanTerm.not(notResult);
+
+        case OR:
+            final Or or = (Or) filter;
+            final BooleanExpression orLeft = convertFilter(or.left());
+            final BooleanExpression orRight = convertFilter(or.right());
+            if (orLeft != null && orRight != null) {
+                return BooleanTerm.or(orLeft, orRight);
+            } else {
+                return null;
+            }
+
+        case AND:
+            final And and = (And) filter;
+            BooleanExpression andLeft = convertFilter(and.left());
+            BooleanExpression andRight = convertFilter(and.right());
+            if (andLeft != null && andRight != null) {
+                return BooleanTerm.and(andLeft, andRight);
+            } else {
+                return null;
+            }
+
+        default:
+            return null;
+        }
+    }
+
+    private ValueExpression[] getMappedLiteralValues(final Object[] values) {
+        return Arrays.stream(values).map(this::getLiteralValue).toArray(ValueExpression[]::new);
+    }
+
+    private ValueExpression getLiteralValue(final Object value) {
+        if (value instanceof Boolean) {
+            return booleanLiteral((Boolean) value);
+        } else if (value instanceof String) {
+            return stringLiteral((String) value);
+        } else if (value instanceof Integer) {
+            return integerLiteral((Integer) value);
+        } else if (value instanceof Long) {
+            return longLiteral((Long) value);
+        } else if (value instanceof Float) {
+            return floatLiteral((Float) value);
+        } else if (value instanceof Double) {
+            return doubleLiteral((Double) value);
+        } else if (value instanceof BigDecimal) {
+            return BigDecimalLiteral.of((BigDecimal) value);
+        } else {
+            return stringLiteral(value.toString());
```

Comment:
Is it possible that `value` is `null`? If yes please add a check to avoid a NullPointerException.

Reply:
Added check and tests